src/collection.ts

```diff
@@ -289,17 +296,31 @@ export class Collection<T extends TigrisCollectionType> {
 		stream.on("end", () => reader.onEnd());
 	}
 
-	findAllStream(reader: ReaderCallback<T>, readFields?: ReadFields) {
-		this.findManyStream(
-			{
-				op: SelectorFilterOperator.NONE,
-			},
-			reader,
-			readFields
-		);
+	search(
+		request: SearchRequest<T>,
+		options?: SearchRequestOptions
+	): Promise<Array<SearchResult<T>>> {
```

Comment:
this should return `Promise<SearchResult>`, how about we make this change in #92 ?